Suspend/resume modules on absence/presence

qistoph opened the pull request:
Add the configurable functionality to suspend all visible modules on absence and resume the modules when a user is present.

qistoph committed:
Suspend/resume modules on absence/presence

qistoph committed:
Delay suspending to match monitor turn off

qistoph committed:
Update deps and add package-lock.json

qistoph committed:
Show presence circle

qistoph committed:
FA user icon

eric-h-st commented on package.json:
```diff
@@ -24,7 +24,7 @@
   "dependencies": {
-    "onoff": "latest",
-    "electron-rebuild": "^1.2.1"
+    "electron-rebuild": "^1.10.1",
```
Why limit the versions?

eric-h-st commented on MMM-PIR-Sensor.js:
```diff
@@ -31,6 +32,25 @@ Module.register('MMM-PIR-Sensor',{
+							clearTimeout(self.suspendTimeout);
+							module.resume();
+						} else { // User not present
+							setTimeout(function() {
```
Why set a separate timeout? Would it not be better to use the timeout defined in `note_helper.js` for reactivation, instead?

eric-h-st reviewed (changes requested):
This is a great idea, but I would suggest to make use of the existing code and timer that counts the same period and is being set and cleared in different scenarios, not covered here. 
Thank you!

qistoph committed:
Add active/inactive classes

qistoph committed:
Update readme about icon

qistoph committed:
Update modules

qistoph committed:
Package updates for latest MM²

qistoph committed:
Link to issue in epoll. Add links

qistoph committed:
Use MagicMirror-rebuild instead of electron-rebuild

https://forum.magicmirror.builders/topic/16074/electron-rebuild-and-magicmirror-v2-18-and-more/

qistoph committed:
Update modules

qistoph committed:
Update dependencies

qistoph committed:
Add MagicMirror-rebuild dependency

qistoph committed:
Update dependencies

qistoph committed:
Update mm-rebuild dependency
